third_party/CMakeLists.txt

```diff
@@ -26,27 +26,44 @@ set(TTMETAL_INCLUDE_DIRS
   ${PROJECT_SOURCE_DIR}/third_party/tt-metal/src/tt-metal/tt_eager
   PARENT_SCOPE
 )
-set(TTMETAL_LIBRARY_DIR ${PROJECT_SOURCE_DIR}/third_party/tt-metal/src/tt-metal-build/lib PARENT_SCOPE)
-set(TTNN_LIBRARY ${TTMETAL_LIBRARY_DIR}/_ttnn.so PARENT_SCOPE)
-set(TTMETAL_LIBRARY -ltt_metal PARENT_SCOPE)
 
-if(EXISTS ${TTNN_LIBRARY})
-message(STATUS "Found TTNN library at ${TTNN_LIBRARY}")
-add_custom_target(tt-metal)
+set(TTMETAL_LIBRARY_DIR ${PROJECT_SOURCE_DIR}/third_party/tt-metal/src/tt-metal-build/lib)
+set(TTNN_LIBRARY_PATH ${TTMETAL_LIBRARY_DIR}/_ttnn.so)
+set(TTMETAL_LIBRARY_PATH ${TTMETAL_LIBRARY_DIR}/libtt_metal.so)
+set(TTEAGER_LIBRARY_PATH ${TTMETAL_LIBRARY_DIR}/libtt_eager.so)
+
+set(TTMETAL_LIBRARY_DIR ${TTMETAL_LIBRARY_DIR} PARENT_SCOPE)
+set(TTNN_LIBRARY_PATH ${TTNN_LIBRARY_PATH} PARENT_SCOPE)
+set(TTMETAL_LIBRARY_PATH ${TTMETAL_LIBRARY_PATH} PARENT_SCOPE)
+set(TTEAGER_LIBRARY_PATH ${TTEAGER_LIBRARY_PATH} PARENT_SCOPE)
+
+if (EXISTS ${TTNN_LIBRARY_PATH} AND EXISTS ${TTMETAL_LIBRARY_PATH} AND EXISTS ${TTEAGER_LIBRARY_PATH})
+  message(STATUS "Found TTNN library at ${TTNN_LIBRARY_PATH}, TTMETAL LIBRARY at ${TTMETAL_LIBRARY_PATH}, TTEAGER LIBRARY at ${TTEAGER_LIBRARY_PATH}")
+  add_custom_target(tt-metal)
 else()
-ExternalProject_Add(
-    tt-metal
-    PREFIX ${TTMLIR_SOURCE_DIR}/third_party/tt-metal
-    CMAKE_GENERATOR Ninja
-    CMAKE_ARGS
-      -DCMAKE_BUILD_TYPE=Release
-      -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
-      -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
-      -DCMAKE_CXX_COMPILER_LAUNCHER=${CMAKE_CXX_COMPILER_LAUNCHER}
-    GIT_REPOSITORY https://github.com/tenstorrent/tt-metal.git
-    GIT_TAG v0.49.0
-    GIT_PROGRESS ON
-)
+  ExternalProject_Add(
+      tt-metal
+      PREFIX ${TTMLIR_SOURCE_DIR}/third_party/tt-metal
+      CMAKE_GENERATOR Ninja
+      CMAKE_ARGS
+        -DCMAKE_BUILD_TYPE=Release
+        -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
+        -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
+        -DCMAKE_CXX_COMPILER_LAUNCHER=${CMAKE_CXX_COMPILER_LAUNCHER}
+      GIT_REPOSITORY https://github.com/tenstorrent/tt-metal.git
+      GIT_TAG v0.49.0
+      GIT_PROGRESS ON
+      BUILD_BYPRODUCTS ${TTNN_LIBRARY_PATH} ${TTMETAL_LIBRARY_PATH} ${TTEAGER_LIBRARY_PATH}
```

Comment:
Oh this is great! I wonder if we need / should have this `if` / `else` at all.  Maybe we should just unconditionally call `ExternalProject_Add`.  I'm not sure why I structured it like that to begin with, but it seems like it'd break dependency tracking if you locally updated the `GIT_TAG` and didn't manually remove the `third_party/` build area.

Reply:
Hey Nick that's a good point, I'll remove the if else!